File: apps/web/src/services/reportService.ts
import type {
  OrderReportDTO,
  RevenueReportDTO,
  UserReportDTO,
  RoomReportDTO,
  ReportOverviewDTO,
  ReportExportRequest
} from '@/types/report'
import { http } from '@/utils/http'

export const reportService = {
  // 获取订单报表
  async getOrderReport(params: {
    startDate: string
    endDate: string
    roomTypeId?: number
    orderStatus?: string
  }): Promise<OrderReportDTO> {
    const response = await http.get('/v1/admin/reports/orders', { params })
    return response.data.data
  },

  // 获取收入报表
  async getRevenueReport(params: {
    startDate: string
    endDate: string
    roomTypeId?: number
  }): Promise<RevenueReportDTO> {
    const response = await http.get('/v1/admin/reports/revenue', { params })
    return response.data.data
  },

  // 获取用户报表
  async getUserReport(params: {
    startDate: string
    endDate: string
  }): Promise<UserReportDTO> {
    const response = await http.get('/v1/admin/reports/users', { params })
    return response.data.data
  },

  // 获取房间报表
  async getRoomReport(params: {
    startDate: string
    endDate: string
    roomTypeId?: number
  }): Promise<RoomReportDTO> {
    const response = await http.get('/v1/admin/reports/rooms', { params })
    return response.data.data
  },

  // 获取报表概览
  async getReportOverview(): Promise<ReportOverviewDTO> {
    const response = await http.get('/v1/admin/reports/overview')
    return response.data.data
  },

  // 导出报表
  async exportReport(request: ReportExportRequest): Promise<string> {
    const response = await http.post('/v1/admin/reports/export', request)
    return response.data.data
  },

  // 刷新报表缓存
  async refreshReportCache(): Promise<void> {
    await http.post('/v1/admin/reports/refresh-cache')
  }
}

File: apps/api/src/main/java/com/hotel/service/ReportService.java
package com.hotel.service;

import com.hotel.dto.report.*;

/**
 * 报表服务接口
 * 提供各种报表的生成和导出功能
 *
 * @author Hotel System
 * @version 1.0
 */
public interface ReportService {

    /**
     * 生成订单报表
     *
     * @param startDate 开始日期
     * @param endDate   结束日期
     * @param roomTypeId 房型ID（可选）
     * @param orderStatus 订单状态（可选）
     * @return 订单报表DTO
     */
    OrderReportDTO generateOrderReport(java.time.LocalDate startDate,
                                      java.time.LocalDate endDate,
                                      Long roomTypeId,
                                      String orderStatus);

    /**
     * 生成收入报表
     *
     * @param startDate 开始日期
     * @param endDate   结束日期
     * @param roomTypeId 房型ID（可选）
     * @return 收入报表DTO
     */
    RevenueReportDTO generateRevenueReport(java.time.LocalDate startDate,
                                         java.time.LocalDate endDate,
                                         Long roomTypeId);

    /**
     * 生成用户报表
     *
     * @param startDate 开始日期
     * @param endDate   结束日期
     * @return 用户报表DTO
     */
    UserReportDTO generateUserReport(java.time.LocalDate startDate,
                                   java.time.LocalDate endDate);

    /**
     * 生成房间报表
     *
     * @param startDate 开始日期
     * @param endDate   结束日期
     * @param roomTypeId 房型ID（可选）
     * @return 房间报表DTO
     */
    RoomReportDTO generateRoomReport(java.time.LocalDate startDate,
                                   java.time.LocalDate endDate,
                                   Long roomTypeId);

    /**
     * 导出报表
     *
     * @param request 导出请求
     * @return 导出文件URL
     */
    String exportReport(ReportExportRequest request);

    /**
     * 获取报表概览数据
     *
     * @return 报表概览DTO
     */
    ReportOverviewDTO getReportOverview();

    /**
     * 刷新报表缓存
     */
    void refreshReportCache();
}